### ISSUE TITLE
Incompatible type error suppressed if function have no return type

### ISSUE THREAD

**Issue by buxx** [python/mypy#6632]
Hello,

I'm reporting a bug (i think it is) about incompatible type error wich is suppressed when container function have no return type. Demonstration:

``` python
class Foo:
    pass


class Bar:
    pass


def get_foo() -> Foo:
    return Foo()


def get_bar() -> Bar:
    return Bar()


def print_foo(foo: Foo) -> None:
    print(foo)


def main():
    foo = get_bar()
    print_foo(foo)


if __name__ == "__main__":
    main()
```

This code send a `Bar` instance to `print_foo`. This is incorrect but mypy don't report:

```
$  mypy t.py
$
```

But, just adding `-> None` to `main` func, so:

``` python
def main() -> None:
    foo = get_bar()
    print_foo(foo)
```
Mypy produce now an error:

```
$  mypy t.py
t.py:23: error: Argument 1 to "print_foo" has incompatible type "Bar"; expected "Foo"
```

Expected behavior should be to got this error with first code example.

```
$ python --version
Python 3.7.1
$ pip freeze | grep mypy
mypy==0.700
mypy-extensions==0.4.1
```

Same behavior with mypy from master (`mypy-0.710+dev.1c0226a50b9ad7db0d4c02b187a6625102031bfc`).

**Comment by buxx** [comment]
Closing because mypy Don't check non typed functions (cf https://mypy.readthedocs.io/en/stable/getting_started.html#function-signatures-and-dynamic-vs-static-typing)
